sys/include/fmt.h

```diff
@@ -76,6 +91,21 @@ size_t fmt_byte_hex(char *out, uint8_t byte);
  */
 size_t fmt_bytes_hex_reverse(char *out, const uint8_t *ptr, size_t n);
 
+/**
+ * @brief Converts a sequence of hex bytes to an array of bytes
+ *
+ * The sequence of hex characters must have an odd length:
```

Comment:
The tests (and your code) suggest that they have to have an *even* length. Also it is undocumented what happens if you provide a string of the wrong length.

Reply:
> The tests (and your code) suggest that they have to have an even length

Indeed, I always do this mistake...